source/agora/common/TransactionPool.d

```diff
@@ -156,17 +153,88 @@ public class TransactionPool
 
         Params:
             tx = the transaction to remove
+            rm_double_spent = remove the TXs that use the same inputs
 
     ***************************************************************************/
 
-    public void remove (const ref Transaction tx) @trusted
+    public void remove (const ref Transaction tx,
+        bool rm_double_spent = true) @trusted
```

Comment:
Are there any use cases for the `rm_double_spent` of `false` value? It would be good to make the concept and code simple. So if there are not, how about removing the parameter for the moment?

Reply:
I had the exact same question.

Reply:
See L179, that call gets redirected to this overload with `rm_double_spent = false`

Reply:
Yeah, right.
But, calling each other `remove` functions is kind of complex to understand. It would be good to make only the second `remove` be able to call the first one. The first one can be called recursively. And then the second `remove` function does not need the `rm_double_spent` parameter, I think. It's just a suggestion.

Reply:
The second one is essentially a wrapper to the first one. So they either both have the `rm_double_spent` or not.

With the original `remove`, we still need to know if we are deleting the TX because it got accepted or because it is now invalid because of another TX that got accepted in to the chain. In the first case, we have to delete the TXs that made other double-spend TXs invalid. In the second case, we just need to delete the TX itself. That is why we need the `rm_double_spent`